### PR TITLE
Release features

### LINKED ISSUE #84: No column headers visible
Column headers have disappeared!

image

### LINKED ISSUE #86: Formatted journal names
At present the journal combo box displays the raw name of the xml file which is fine, but isn't particularly visually appealing!

Reformatting, for instance, `journal_21_1.xml` to show `Cycle 21/1` would look a lot better and put the naming into a format which is generally accepted as being the norm around ISIS.

### LINKED ISSUE #88: Use proper capitalisation scheme in instrument dropdown
We should honour the accepted capitalisation of instrument names in the instrument selector dropdown.  This information could be readily stored as "DisplayName" in the instrument xml file.

Most instrument names are straight capitalisations, but there are some exceptions: `SANS2d`, `ChipIr`. There may be others, but we will have to wait for the instrument scientists to tell us what they are / prefer!

### LINKED ISSUE #89: Tweak the Find dialog
The `Find` dialog can be improved by changing / adding some text:

1. The dialog title is currently "Enter search query" - really I think that just "Find" is enough.
2. The dialog text is currently "search runs for:" but we should probably expand this to make it clear what is being searched. Suggestion: "Find in current run data (RB, user, title):"
3. I also think that the `Find` menu should be renamed to `Search` as that feels more encompassing of the available actions.

### LINKED ISSUE #90: Find next / previous doesn't wrap
Find next (`F3`) and find previous (`Shift-F3`) don't wrap around to the top/bottom when reaching the last/first match.

### LINKED ISSUE #92: Improved date formatting
The run start / end time format stored in the journals (e.g. `2022-02-03T10:45:23`) is a bit ugly for humans and we could consider reformatting this into a more acceptable form - e.g. `03/02/2022 10:45:23`.  For times that started "today" or "yesterday", we could display as such (e.g. `Today 10:45:23`).

### LINKED ISSUE #93: Zoom range coordinates not showing
In the current version the feature(s) added in #69 seem to have disappeared - for instance the coordinate labels to show the extrema of zoom ranges are no longer shown.

### PR DESCRIPTION
**Pull request by EthanDevlinSTFC**
A variety of minor issues have been covered here. Broadly, each commit covers one of yesterdays new issues and are titled fairly clearly so individual code review can be assessed per commit for legibility. Had to merge the last PR into this so hopefully everything is fine there.

closes #84 
closes #86
closes #88
closes #89 
closes #90 
closes #92 
closes #93

### CODE REVIEW

File: frontend/datagathering.cpp [hunk 1]
```diff
@@ -157,19 +160,20 @@ void MainWindow::currentInstrumentChanged(const QString &arg1)
...
+    auto currentText = ui_->cyclesBox->itemText(index);
+    ui_->cyclesBox->setDisabled(currentText.isEmpty());
+    ui_->filterBox->setDisabled(currentText.isEmpty());
+    if (currentText.isEmpty())
         return;
 
-    if (arg1[0] == '[')
+    if (currentText[0] == '[')
     {
         for (auto tuple : cachedMassSearch_)
```
Comment (trisyoungs) [inline frontend/datagathering.cpp]: Could possibly use `std::find_if` and structured bindings here to neaten things up (untested):
```
auto it = std::find_if(cachedMassSearch_.begin(), cachedMassSearch_.end(), [currentText](const auto &&[worker, name]) { return name == currentText.mid(1, currentText.length() - 2); } );
```

File: extra/instrumentData.xml [hunk 2]
```diff
@@ -3,146 +3,182 @@
 	<inst name="alf">
 		<location>ndxalf</location>
 		<type>Neutron</type>
+		<displayName>alf</displayName>
```
Comment (trisyoungs) [inline extra/instrumentData.xml]: As alluded to previously, all the `displayName`s can be fully capitalised in the first instance, with the exceptions of `SANS2d` and `ChipIr`.

File: frontend/searching.cpp [hunk 3]
```diff
@@ -121,7 +121,7 @@ void MainWindow::selectSimilar()
 }
 void MainWindow::on_actionSearch_triggered()
 {
-    QString textInput = QInputDialog::getText(this, tr("Enter search query"), tr("search runs for:"), QLineEdit::Normal);
+    QString textInput = QInputDialog::getText(this, tr("Find"), tr("Find in current run data (RB, user, title,...):"), QLineEdit::Normal);
```
Comment (trisyoungs) [inline frontend/searching.cpp]: Question - are these the only fields that are searched in the backend? If so the ellipsis can be removed. If the list isn't much longer, those keys can be added to the list.

File: frontend/searching.cpp [hunk 4]
```diff
@@ -62,6 +62,8 @@ void MainWindow::findDown()
     {
         if (currentFoundIndex_ < foundIndices_.size() - 1)
             currentFoundIndex_ += 1;
+        else
+            currentFoundIndex_ = 0;
```
Comment (trisyoungs) [inline frontend/searching.cpp]: Possible one-liner here (but best check it!):
```
currentFoundIndex_ = ++currentFoundIndex_ % foundIndices_.size();
```

File: frontend/jsontablemodel.cpp [hunk 5]
```diff
@@ -119,6 +119,8 @@ QVariant JsonTableModel::data(const QModelIndex &index, int role) const
                                        QString::number(minutes).rightJustified(2, '0') + ":" +
                                        QString::number(seconds).rightJustified(2, '0'));
                     }
+                    if(QDateTime::fromString(v.toString(), "yyyy-MM-dd'T'HH:mm:ss").isValid())
+                        return QDateTime::fromString(v.toString(), "yyyy-MM-dd'T'HH:mm:ss").toString("dd/MM/yyyy HH:mm:ss");
```
Comment (trisyoungs) [inline frontend/jsontablemodel.cpp]: I was going to complain here that you didn't add specific "Today" and "Yesterday" cases, but obviously that's hard to test until we start cycle!  Can make a note to come back to this issue.

File: frontend/chartview.cpp [hunk 6]
```diff
@@ -157,19 +158,12 @@ void ChartView::mouseMoveEvent(QMouseEvent *event)
...
-            maxY = yVal;
-            minY = yVal;
-        }
-        else
-        {
-            QValueAxis *yAxis = qobject_cast<QValueAxis *>(chart()->axes(Qt::Vertical)[0]);
-            maxY = yAxis->max();
-            minY = yAxis->min();
-        }
+        QValueAxis *yAxis = qobject_cast<QValueAxis *>(chart()->axes(Qt::Vertical)[0]);
```
Comment (trisyoungs) [inline frontend/chartview.cpp]:
```suggestion
        auto *yAxis = qobject_cast<QValueAxis *>(chart()->axes(Qt::Vertical)[0]);
```

File: frontend/datagathering.cpp [hunk 7]
```diff
@@ -28,7 +28,11 @@ void MainWindow::handle_result_instruments(HttpRequestWorker *worker)
         {
             // removes header_ file
             if (value.toString() != "journal.xml")
-                ui_->cyclesBox->addItem(value.toString());
+            {
+                auto displayName =
+                    "cycle " + value.toString().split("_")[1] + "/" + value.toString().split("_")[2].remove(".xml");
```
Comment (trisyoungs) [inline frontend/datagathering.cpp]:
```suggestion
                    "Cycle " + value.toString().split("_")[1] + "/" + value.toString().split("_")[2].remove(".xml");
```

**Review by trisyoungs — CHANGES_REQUESTED** [review]
Lots of good stuff in there. Just a few small things to fixup.

File: frontend/searching.cpp (hunk 3 above)
Comment (EthanDevlinSTFC) [reply frontend/searching.cpp]: It searches every currently visible column

File: frontend/jsontablemodel.cpp (hunk 5 above)
Comment (EthanDevlinSTFC) [reply frontend/jsontablemodel.cpp]: I can implement this for now and then when I get round to unit testing we should be able to assess this

**Review by trisyoungs — APPROVED** [review]
All looking good! :+1:

**Pull request merged**